File: Sample-Microservices/dotnet/Xerris.SRE.ProductCatalog/src/Xerris.SRE.ProductCatalog/Controllers/ProductsController.cs
using Microsoft.AspNetCore.Mvc;
using Serilog;
using Xerris.SRE.ProductCatalog.Models;
using Xerris.SRE.ProductCatalog.Repositories;
using Xerris.SRE.ProductCatalog.Services;

namespace Xerris.SRE.ProductCatalog.Controllers;

[ApiController]
[Route("[controller]")]
public class ProductsController : Controller
{
    private readonly IProductService service;
    
    public ProductsController(IProductService service)
    {
        this.service = service;
    }

    [HttpGet(Name = "GetProducts")]
    public async Task<IEnumerable<Product>> GetAllProducts()
    {
        Log.Information("Fetching all Products");
        return await service.GetAllProducts();
    }
    
    [HttpGet("/{sku:guid}",Name = "GetProductBySku")]
    public async Task<Product> GetBySku(Guid sku)   
    {
        Log.Information("Fetching Product {@Sku}", sku);
        return await service.FindBySku(sku);
    }
}

File: Sample-Microservices/NodeJS/product-catalog/src/products/products.controller.ts
import { Controller, Get } from '@nestjs/common';
import { ProductsService } from './products.service';
import { ConfigService } from '@nestjs/config';
import { Product } from "../models/product.model";

@Controller('products')
export class ProductsController {
  constructor(private productsService: ProductsService,private config: ConfigService) {}

  @Get()
  getAllProducts(): Product[] {
    const pricingUrl = this.config.get<string>('pricing-url');
    console.log(pricingUrl);
    return this.productsService.GetAllProducts();
  }
}
